### DISCUSSION TITLE
Installation error on Emacs 29.1 on Mac OS X Ventura 13.4.1

### DISCUSSION THREAD

**Discussion by wbl** [magit/magit#4984] [Support]
On installing magit from package-list after deleting entirely, I get the following error 
ld: library not found for -lSystem.

This is the root cause of a volumninous and uninteresting backtrace I am happy to provide should it prove helpful. This could easily be an emacs issue, but I really don't know.

**Comment by tarsius** [comment]
Sounds like a linker doesn't find a library. `magit-libgit.el` uses a c library but because this is only a stub and might cause issues such as this, that elisp library is not distributed with the packages from NonGNU ELPA and MELPA.

Maybe you have not run `package-refresh-contents` in a very long time and so Package attempts to install an old version of Magit, which still contains that useless elisp library.

**Reply by wbl** [reply]
I'm seeing 3.3.0 as the version in the package list, let me try again.

**Reply by wbl** [reply]
Error still happens afterwards: ran package-refresh-contents, uninstall and reinstall. It's downloaded from elpa.

**Reply by tarsius** [reply]
I've just downloaded both https://elpa.nongnu.org/nongnu/magit-3.3.0.tar and https://stable.melpa.org/packages/magit-3.3.0.tar, and neither one of them contains `magit-libgit.el`. Try removing a locally cached tarball (somewhere in `~/.emacs.d/elpa` I would assume) before installing again.

**Reply by wbl** [reply]
I've blown away ~/.emacs.d a few times and reinstalled emacs in between (with disabling native compilation) and still no luck. Maybe I'll try with a non-macports one, see if that's the bug.

**Reply by wbl** [reply]
Sorry, this was a macports bug! All is good with a different build
